RabbitMq/BatchConsumer.php

```diff
@@ -359,7 +372,7 @@ protected function maybeStopConsumer()
             pcntl_signal_dispatch();
         }
 
-        if ($this->forceStop) {
+        if ($this->forceStop || ($this->batchAmount == $this->target && $this->target > 0)) {
```

Comment:
When will this trigger? given the fact batchAmount only gets incremented when a consume call is finished AND target gets rewritten for every consume() call?

Reply:
This is exactly the point. target will be set from BatchConsumerCommand. When a batch of messages are consumed, then $this->batchAmount is incremented and if a number ($this->target) of batches set from command are consumed, then the consumer can be stopped.

Reply:
I agree but why set the target only on consume calls, especially since $this->batchAmount only gets incremented by 1 every time consume() finishes?